### PR TITLE
Updated output colors

### PR DESCRIPTION
**Pull request by vanowm**
* All regexes derived from lexilla's LexErrorList.cxx
* fixed Colorization issues #120

### CODE REVIEW

**Comment by vanowm** [comment]
The scope names are taken from LexErrorList.cxx.
Most colors are useless for AutoIt, but It's close to output in SciTe

Testing it against
```autoit
; colors In SciTE are definied in   others.properties   for example like this: style.errorlist.3=fore:#FF000

ConsoleWrite('Default text |SCE_ERR_DEFAULT|style.errorlist.0=fore:#000000' & @CRLF) ; black text color             => style.errorlist.0
ConsoleWrite( '> |first character `>` |SCE_ERR_CMD|style.errorlist.4=fore:#0000FF' & @CRLF);
ConsoleWrite( '< |first character `<` |SCE_ERR_DIFF_DELETION|style.errorlist.12=fore:#FF8800,bold' & @CRLF)
ConsoleWrite( '! |first character `!` |SCE_ERR_DIFF_CHANGED|style.errorlist.10=fore:#F70000,bold' & @CRLF)
ConsoleWrite( '+ |first character `+` |SCE_ERR_DIFF_ADDITION|style.errorlist.11=fore:#007F00,bold' & @CRLF)
ConsoleWrite( '- |first character `-` |SCE_ERR_DIFF_DELETION|style.errorlist.12=fore:#FF8800,bold' & @CRLF)
ConsoleWrite( '+++ |first characters `+++ ` |SCE_ERR_DIFF_MESSAGE|style.errorlist.13=fore:#7F0000' & @CRLF)
ConsoleWrite( '--- |first characters `--- ` |SCE_ERR_DIFF_MESSAGE|style.errorlist.13=fore:#7F0000' & @CRLF)
ConsoleWrite( 'cf90- |first characters `cf90-` |SCE_ERR_ABSF|style.errorlist.18=fore:#FF0000 //Absoft Pro Fortran 90/95 v8.2 error and/or warning message' & @CRLF)
ConsoleWrite( 'fortcom: |first characters `fortcom:` |SCE_ERR_IFORT|style.errorlist.17=fore:#FF0000 //Intel Fortran Compiler v8.0 error/warning message' & @CRLF)
ConsoleWrite( 'File ", line |contains string `File "` and `, line` (order doesn''t matter) |SCE_ERR_PYTHON|style.errorlist.1=fore:#FF0000' & @CRLF)
ConsoleWrite( ' in on line |contains string ` in ` and  ` on line ` (order doesn''t matter) |SCE_ERR_PHP|style.errorlist.14=fore:#FF0000' & @CRLF)
ConsoleWrite( 'Error at () : |first characters `Error ` followed by ` at (` followed by `) : ` |SCE_ERR_IFC|style.errorlist.16=fore:#FF0000 //Intel Fortran Compiler error/warning message' & @CRLF)
ConsoleWrite( 'Warning at () : |first characters `Warning ` followed by ` at (` followed by `) : ` |SCE_ERR_IFC|style.errorlist.16=fore:#FF0000 //Intel Fortran Compiler error/warning message' & @CRLF)
ConsoleWrite( 'Error |first characters `Error ` |SCE_ERR_BORLAND|style.errorlist.5=fore:#B06000 //Borland error message' & @CRLF)
ConsoleWrite( 'Warning |first characters `Warning ` |SCE_ERR_BORLAND|style.errorlist.5=fore:#B06000 //Borland warning message' & @CRLF)
ConsoleWrite( 'at line file |contains string `at line ` and `file ` (order doesn''t matter) |SCE_ERR_LUA|style.errorlist.8=fore:#FF0000 //Lua 4 error message' & @CRLF)
ConsoleWrite( ' at   line |contains string ` at ` plus at least 1 character followed by ` line ` |SCE_ERR_PERL|style.errorlist.6=fore:#FF0000 //perl error message: <message> at <file> line <line>' & @CRLF)
ConsoleWrite( '   at :line |first characters `   at ` followed by `:line` |SCE_ERR_NET|style.errorlist.7=fore:#FF0000 //A .NET traceback' & @CRLF)
ConsoleWrite( 'Line , file |first characters `Line ` followed by `, file ` |SCE_ERR_ELF|style.errorlist.15=fore:#FF0000 // Essential Lahey Fortran error message' & @CRLF)
ConsoleWrite( 'line column |first characters `line ` followed by ` |SCE_ERR_TIDY|style.errorlist.19=fore:#FF0000 //HTML tidy style: line 42 column 1' & @CRLF)
ConsoleWrite( '	at (.java: |first characters `\tat ` followed by `(` and `.java:` |SCE_ERR_JAVA_STACK|style.errorlist.20=fore:#FF0000 //Java stack back trace' & @CRLF)
ConsoleWrite( 'In file included from |first characters `In file included from ` |SCE_ERR_GCC_INCLUDED_FROM|style.errorlist.22=fore:#800080 //GCC showing include path to following error' & @CRLF)
ConsoleWrite( '                 from |first characters `                 from ` |SCE_ERR_GCC_INCLUDED_FROM|style.errorlist.22=fore:#800080 //GCC showing include path to following error' & @CRLF)
ConsoleWrite( 'NMAKE : fatal error |first characters `NMAKE : fatal error` |SCE_ERR_MS|style.errorlist.3=fore:#808000 // Microsoft nmake fatal error: NMAKE : fatal error <code>: <program> : return code <return>' & @CRLF)
ConsoleWrite( 'warning LNK |contains string `warning LNK` |SCE_ERR_MS|style.errorlist.3=fore:#808000 //Microsoft linker warning: {<object> : } warning LNK9999' & @CRLF)
ConsoleWrite( ' |  |contains string ` | ` |SCE_ERR_GCC_EXCERPT[25] (affected by |style.errorlist.32=fore:#B06000,$(font.small)?) // GCC code excerpt and pointer to issue:  |   ^~~~~~~~~~~~' & @CRLF)
ConsoleWrite( ' |+ |contains string ` |+` |SCE_ERR_GCC_EXCERPT[25] (affected by |style.errorlist.32=fore:#B06000,$(font.small)?) // GCC code excerpt and pointer to issue:  |   ^~~~~~~~~~~~' & @CRLF)
ConsoleWrite( 'foo	bar	123blah|starts with non-whitespace character <TAB><anything or nothing><TAB><NUMBER><anything>| /\S\t.*\t\d+.*/ |SCE_ERR_CTAG| style.errorlist.9=fore:#FF00FF' & @CRLF)
ConsoleWrite( '' & @CRLF)

ConsoleWrite('! = Bold Red' & @CRLF)      ; ! = red bold text color      => style.errorlist.10
ConsoleWrite('> = Blue' & @CRLF)     ; > = blue text color          => style.errorlist.4
ConsoleWrite('+ = Green' & @CRLF)    ; + = green text color         => style.errorlist.11
ConsoleWrite('< = Orange' & @CRLF)   ; < = orange bold text color   => style.errorlist.12
ConsoleWrite('- = Orange' & @CRLF)   ; - = orange bold text color   => style.errorlist.12
ConsoleWrite('--- = Brown' & @CRLF)  ; --- = brown text color       => style.errorlist.13
ConsoleWrite('+++ = Brown' & @CRLF)  ; +++ = brown text color       => style.errorlist.13

ConsoleWrite('(5) : = Red (jump to line 5 when double-clicked)' & @LF)
ConsoleWrite('Start with String or Integer then ' & @TAB & '6' & ' = Pink (jump to line 6 when double-clicked)' & @LF)
ConsoleWrite('(' & @ScriptLineNumber & ') : = Red (jump to line ' & @ScriptLineNumber & ' when double-clicked)' & @CRLF)


; Red text color (when double-clicked jump to specified line in specified script file) => style.errorlist.3
; ConsoleWrite('FileFullPath' ( #ScriptLineNumber ) : 'AnyDescription=CallTipToPopUp')
ConsoleWrite(@ScriptFullPath & '(' & @ScriptLineNumber+2 & ') : Description=CallTip to PopUp' & @CRLF)

; Red text color (when double-clicked jump to specified line in specified script file) => style.errorlist.3
; ConsoleWrite('Description' & ' in ' & 'FileFullPathOrFileName' & ' ( #ScriptLineNumber ) : ' & 'AnyDescription=CallTipToPopUp')
ConsoleWrite('IMPORTANT TODO in ASin.au3 on line 5 SOME Description' & @CRLF)

; Pink text color (when double-clicked jump to specified line in specified script file) => style.errorlist.9
; ConsoleWrite('Description_WithOut_Space' & @TAB & 'FileFullPathOrFileName' & @TAB & #ScriptLineNumber & 'Any other Descripion')
ConsoleWrite('DescripionForPink' & @TAB & 'ACos.au3' &  @TAB & '10' & ' = SOME Description for Pink text color' & @CRLF)

; Violet+Black text color (when double-clicked jump to specified line in specified script file) => style.errorlist.8
; ConsoleWrite(@TAB & 'FileFullPathOrFileName' & @TAB & #ScriptLineNumber & 'Any other Descripion')
ConsoleWrite(@TAB & 'ATan.au3' &  ':10:' & ' SOME Description for Violet+Black text color' & @CRLF)

; Chocolate text color => style.errorlist.5
; ConsoleWrite('Warning' & ' ' & 'Any kind of description')
ConsoleWrite('Warning ' & 'Some description for Chocolate text color' & @CRLF)

; Chocolate text color => style.errorlist.5
; ConsoleWrite('Error' & ' ' & 'Any kind of description')
ConsoleWrite('Error ' & 'Some description for Chocolate text color' & @CRLF)

; not jet checked
ConsoleWrite('In file included from SOME Description' & @CRLF) ; to check
ConsoleWrite('line 10 column 20' & @CRLF) ; to check

;Purple for trailing operator :#:
ConsoleWrite('Purple for any line ending with ":number:". Add as much space to the end as you want.                   :' & @ScriptLineNumber & ':' & @CRLF)


Consolewrite("Port:	4444" & @CRLF)
Consolewrite("	123" & @CRLF)
```

**Comment by Danp2** [comment]
> Start with String or Integer then 	6 = Pink (jump to line 6 when double-clicked)

> DescripionForPink	ACos.au3	10 = SOME Description for Pink text color

FWIW, the first line doesn't appear pink in Scite, nor does it jump to the designated line. The second one does for me.

**Comment by vanowm** [comment]
Yeah, these are collected from autoit [forum](https://www.autoitscript.com/forum/topic/125490--), by default not all matched keywords have assigned color in SciTe.
And perhaps these changed between different versions of SciTe

**Comment by vanowm** [comment]
There is seems to be a [bug](https://github.com/microsoft/vscode/issues/162626) (I think) that doesn't match anything when `^` used in regex and a line above is empty

**Comment by Danp2** [comment]
https://github.com/loganch/AutoIt-VSCode/blob/a6ddc236281f0d388c1003172eda446c64c9d545/syntaxes/vscode-autoit-output.tmLanguage.json#L62-L64

I thought this would work since you switched from `^` to `\GG`, but it isn't displaying in red for me.

**Comment by Danp2** [comment]
@vanowm It seems like it was working correctly when a blank line was present prior to this PR, so I suspect the issue is with the switch to using \\G instead of ^.  From [here](https://www.apeth.com/nonblog/stories/textmatebundle.html) --

> \G — This is a way of anchoring an attempted match in exactly the place where the previous match ended (useful when a grammar works by carving up a document completely into contiguous scopes)

Is that the behavior that you want / expect? For that to work, I would think it would require a separate entry to match the blank line.

**Comment by vanowm** [comment]
No, it's when `^` followed by optional character used in `begin` regex, see [this](https://github.com/microsoft/vscode/issues/162626) bug report

The `\G` anchor means beginning of text between `begin` and `end` matches

The reason this PR uses `begin`/`end` keys, is to eliminate duplications of 
```
^(?:(\\d{2}:\\d{2}:\\d{2}\\.\\d{3})?( ?#\\d+:| {3,})? )?
```
which otherwise would have to be in each regex.

File: syntaxes/vscode-autoit-output.tmLanguage.json [hunk 1]
```diff
@@ -3,116 +3,167 @@
...
-      }
+    {"include": "#header"}
+  ],
+  "repository": {
+    "header": {
+      "patterns": [
+        {
+          "comment": "Timestamp and/or process ID (12:34:56.789 #123: )",
+          "begin": "^(?:(\\d{2}:\\d{2}:\\d{2}\\.\\d{3})?( ?#\\d+:| {3,})? )?",
+          "end": "\\r?\\n|$",
```
Comment (RedCMD) [inline syntaxes/vscode-autoit-output.tmLanguage.json:14]: The header rule ends itself on a newline
so if nothing on same line pushes it out
then nothing will match on the next line

File: syntaxes/vscode-autoit-output.tmLanguage.json [hunk 2]
```diff
@@ -3,116 +3,167 @@
...
+            "3": { "name": "vscode-autoit-output-process-pid" }
+          }
+        },
+        {
+          "comment": "exit with no error code (0 or -1)",
+          "match": "\\G>Exit code (?:0|-1) .*",
+          "name": "vscode-autoit-output-EXIT"
+        },
+        {
+          "match": "\\G>Exit code 1 .*",
```
Comment (RedCMD) [inline syntaxes/vscode-autoit-output.tmLanguage.json:44]: `(\\G|^)`

File: syntaxes/vscode-autoit-output.tmLanguage.json (hunk 1 above)
Comment (vanowm) [reply syntaxes/vscode-autoit-output.tmLanguage.json:14]: that's what happens with `$` by itself

File: syntaxes/vscode-autoit-output.tmLanguage.json (hunk 2 above)
Comment (vanowm) [reply syntaxes/vscode-autoit-output.tmLanguage.json:44]: the`↑` doesn't match anything in nested patterns...

Comment (RedCMD) [reply syntaxes/vscode-autoit-output.tmLanguage.json:44]: your nested pattern doesn't go over a newline for it to match
it ends before it

Comment (vanowm) [reply syntaxes/vscode-autoit-output.tmLanguage.json:44]: yes, that's why it was replaced with `\\G`, but you commented with `(\\G|^)`, as if `^` would do anything...

Comment (RedCMD) [reply syntaxes/vscode-autoit-output.tmLanguage.json:44]: ^ will only work if something pushes the end rule out
currently you have nothing that does that

Comment (RedCMD) [reply syntaxes/vscode-autoit-output.tmLanguage.json:44]: is there an example file I can look at?
to see what possible solutions there are

Comment (vanowm) [reply syntaxes/vscode-autoit-output.tmLanguage.json:44]: This is the text I'm testing it against, and image of SciTe editor output that I'm trying make this extension to match to
[output.zip](https://github.com/loganch/AutoIt-VSCode/files/9738193/output.zip)
So far, it's working 100%

**Pull request merged**